proposals/0030-core-image-story.md

```diff
@@ -0,0 +1,427 @@
+
+## Image component
+
+A component that outputs a simple `<img> `tag with all the required attributes needed to show the image.
+
+### Properties
+
+#### width and height
+
+Those properties are used for resizing the image and ensuring there's no CLS. In most cases, the `width` and `height` would be automatically inferred from the source file and not passing them would just result in the same dimensions.
+
+The only time those properties are required is for remote images.
+
+#### format
+
+`format` is used to set the resulting wanted format for the image.
+
+Default is left to the services to choose. But for the services exposed by Astro, this property is always optional and is set to `webp`.
+
+#### quality
+
+`quality` inform the quality to use when transforming the image. It can either be a number from `0` to `100`, or a preset from `low`, `medium`, `high`, `max`. Specific implementation here is left to the image service used, as they do not necessarily all abide by the same rules.
+
+Default is left to the services to choose. But for the services exposed by Astro, this property is always optional.
+
+#### alt
+
+`alt` is a required property in all cases when using the component. It can be explicitly set to `""`, for cases where an alt-text is not required.
+
+### Facts
+
+- For the services exposed by Astro: For ESM images, the only required property is `src`. Remote images (ex: `http://example.com/image.png`, `/image.png` or `${import.meta.env.BASE_URL}/image.png`) however require `width` and `height` to be set manually.
+- `src` can be a dynamic import, but be aware that it must respect [Vite's limitations on dynamic imports](https://vitejs.dev/guide/features.html#dynamic-import)
+
+## JavaScript API
+
+A `getImage` function taking the same parameters as the image component is also supported, for use cases inside the frontmatter or outside `img` tag (ex: `background-image`)
+
+This function returns an object with all the properties needed to use / show an image.
+
+```ts
+// Example interface describing the content of `myImage`
+interface getImageResult {
+  // Contain the original options passed to `getImage`
+  options: Record<string, any>;
+  // Contain a path you can use to render the image
+  src: string;
+  // Contain additional HTML attributes needed to render the image (ex: `width`, `height`, `loading`)
+  attributes: Record<string, any>;
+}
+```
+
+> Specific types not necessarily accurate to implementation. A more or less open-ended type for `options` and `attributes` is nonetheless required, as different services are allowed to return different things (including non-standard HTML attributes.)
+
+## Image Services
+
+By default, Astro will ship with two services can users can choose to transform their images: Squoosh, and Sharp. Both services more or less offer the same results and mostly differ in where they can run. Squoosh will be the default, because albeit slower than Sharp, it supports more platforms.
```

Comment:
```suggestion
By default, Astro will ship with two services that users can choose from to transform their images: Squoosh, and Sharp. Both services more or less offer the same results and mostly differ in where they can run. Squoosh will be the default, because albeit slower than Sharp, it supports more platforms.
```

Are there links we can provide for these two services?